File: src/main/java/cairo/CairoFileType.java
package cairo;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class CairoFileType extends LanguageFileType {
	public static final CairoFileType INSTANCE = new CairoFileType();

	private CairoFileType() {
		super(CairoLanguage.INSTANCE);
	}

	@NotNull
	@Override
	public String getName() {
		return "Cairo File";
	}

	@NotNull
	@Override
	public String getDescription() {
		return "Cairo language source code";
	}

	@NotNull
	@Override
	public String getDefaultExtension() {
		return "cairo";
	}

	@Nullable
	@Override
	public Icon getIcon() {
		return CairoIcons.ICON_CAIRO_16;
	}
}

File: src/main/kotlin/cairoplugin/language/CairoFileType.kt
package cairoplugin.language

import com.intellij.openapi.fileTypes.LanguageFileType
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable
import javax.swing.Icon




class CairoFileType: LanguageFileType(CairoLanguage.INSTANCE)  {
    companion object {
        val INSTANCE: CairoFileType = CairoFileType()
    }

    @NotNull
    override fun getName(): String {
        return "Cairo File"
    }

    @NotNull
    override fun getDescription(): String {
        return "Cairo language file"
    }

    @NotNull
    override fun getDefaultExtension(): String {
        return "cairo"
    }

    @Nullable
    override fun getIcon(): Icon {
        return CairoIcons.FILE
    }
}